api/src/main/java/org/pmiops/workbench/institution/InstitutionServiceImpl.java

```diff
@@ -0,0 +1,146 @@
+package org.pmiops.workbench.institution;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Collectors;
+import java.util.stream.StreamSupport;
+import javax.inject.Provider;
+import org.pmiops.workbench.db.dao.InstitutionDao;
+import org.pmiops.workbench.db.dao.InstitutionEmailAddressDao;
+import org.pmiops.workbench.db.dao.InstitutionEmailDomainDao;
+import org.pmiops.workbench.db.model.DbInstitution;
+import org.pmiops.workbench.db.model.DbInstitutionEmailAddress;
+import org.pmiops.workbench.db.model.DbInstitutionEmailDomain;
+import org.pmiops.workbench.db.model.DbStorageEnums;
+import org.pmiops.workbench.model.Institution;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+@Service
+public class InstitutionServiceImpl implements InstitutionService {
+
+  private final Provider<InstitutionDao> institutionDaoProvider;
+  private final Provider<InstitutionEmailDomainDao> institutionEmailDomainDaoProvider;
+  private final Provider<InstitutionEmailAddressDao> institutionEmailAddressDaoProvider;
+
+  @Autowired
+  InstitutionServiceImpl(
+      Provider<InstitutionDao> institutionDaoProvider,
+      Provider<InstitutionEmailDomainDao> institutionEmailDomainDaoProvider,
+      Provider<InstitutionEmailAddressDao> institutionEmailAddressDaoProvider) {
```

Comment:
Why are using Provider and not injecting the dao directly? Injecting the dao directly should be sufficient.

Reply:
Out of ignorance.  Updated.